packages/office-ui-fabric-react/src/components/Button/examples/BaseButton.tsx

```diff
@@ -27,9 +26,12 @@ export const BaseButton: React.FunctionComponent<IBaseButtonProps> = props => {
     </>
   );
 
-  return (
-    <Root {...root} {...rest} className={rootClassName}>
-      {content}
-    </Root>
-  );
+  const { htmlType, propertiesType } = _deriveRootType(props);
+  const rootProps = { ...getNativeProps(rest, propertiesType), type: htmlType, className: rootClassName };
+
+  return <Root {...rootProps}>{content}</Root>;
```

Comment:
Can we do this conditionally? If it is a primitive type, that this is correct, if it is a component, spread all props